quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -106,6 +121,7 @@ impl IngestController {
             metastore,
             ingester_pool,
             replication_factor,
+            rebalance_permits: Arc::new(Semaphore::new(1)),
```

Comment:
nitpick:

I suspect a Mutex and OwnedMutexGuard would have better captured your intent.